Document/0x05b-Basic-Security_Testing.md

```diff
@@ -74,58 +74,129 @@ As mentioned before, starting with Android 7, the Android OS will no longer trus
 
-res/xml/network_security_config.xml:
-```xml
-<?xml version="1.0" encoding="utf-8"?>
-<network-security-config>
-    <base-config>
-        <trust-anchors>
-            <certificates src="@raw/extracas"/>
-            <certificates src="system"/>
-        </trust-anchors>
-    </base-config>
-</network-security-config>
-```
-The apps must include an entry in the manifest file to point to the network security configuration file.
 ```xml
 <?xml version="1.0" encoding="utf-8"?>
 <manifest ... >
     <application android:networkSecurityConfig="@xml/network_security_config"
                     ... >
         ...
-    </application>decomp
+    </application>
 </manifest>
 ```
 
-In order to intercept the traffic of an application running on Android 7.0 and higher, you must follow the steps below:
+The network security configuration uses an XML file where the app specifies which CA certificates will be trusted. There are various ways to bypass the Network Security Configuration, which will be described below. Please also see the [Security Analyst’s Guide to Network Security Configuration in Android P](https://www.nowsecure.com/blog/2018/08/15/a-security-analysts-guide-to-network-security-configuration-in-android-p/ "Security Analyst’s Guide to Network Security Configuration in Android P") for further information.
 
-- Decompile the app using decompilation tools.[Manual static Analysis] provides details about decompiling (https://github.com/OWASP/owasp-mstg/blob/master/Document/0x05b-Basic-Security_Testing.md#manual-static-analysis)
-- Make the application trust the proxy's certificate by creating a network security configuration with the giving certificate as explained above
-- Repackage the app. The [Android developer documentation](https://developer.android.com/studio/publish/app-signing#signing-manually) explains how it's done.
+##### Adding the User Certificates to the Network Security Configuration
+
+There are different configurations available for the Network Security Configuration to [add non-system Certificate Authorities](https://developer.android.com/training/articles/security-config#CustomTrust "Custom Trust") via the src attribute:
+
+```xml
+<certificates src=["system" | "user" | "raw resource"]
+              overridePins=["true" | "false"] />
+```
+Each certificate can be one of the following:
+- a "raw resource" ID pointing to a file containing X.509 certificates
+- "system" for the pre-installed system CA certificates
+- "user" for user-added CA certificates
+
+
+The CA certificates trusted by the app can be a system trusted CA as well as a user CA. Usually you will have added the certificate of your interception proxy already as additional CA in Android. Therefore we will focus on the "user" setting, which allows you to force the Android app to trust this certificate with the following Network Security Configuration configuration below:
```

Comment:
Isn't this explained in the network security part of the android part of MSTG as well? possibly it might be a better idea to refer to that.

Reply:
You are right, its a bit redundant. But I think it should be emphasised in both sections as this is quite important to understand instead of referencing. 
But I also cleaned a bit the networks security configuration part in 0x05g and I think it's now better aligned and I am referencing from 0x05g to 0x05b on how to bypass/patch it.